### PR TITLE
Group all init of Legend.legendPatch together.

### PR DESCRIPTION
**Pull request by anntzer**
Seems easier to follow that spreading the init of legendPatch across
`__init__` and intertwined with other things.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/legend.py [hunk 1]
```diff
@@ -483,45 +483,32 @@ def __init__(self, parent, handles, labels,
...
+                      else "square,pad=0"),
+            mutation_scale=self._fontsize,
+            snap=True,
+        )
         self._set_artist_props(self.legendPatch)
 
-        self._drawFrame = frameon
         if frameon is None:
-            self._drawFrame = rcParams["legend.frameon"]
+            frameon = rcParams["legend.frameon"]
```
Comment (timhoffm) [inline lib/matplotlib/legend.py]:
~~~
        self._drawFrame = (frameon if frameon is not None
                           else rcParams["legend.frameon"])
~~~
?

Comment (anntzer) [reply lib/matplotlib/legend.py]: sure...

**Comment by QuLogic** [comment]
Seems to have broken: `rcParams` is undefined.

**Comment by anntzer** [comment]
ah, prs crossing one another... another try.

**Pull request merged**